### PR TITLE
[ONNX] Tool to serialize ONNX models

### PR DESCRIPTION
**Pull request by postrational**
Simple tool to use for debugging problems with ONNX models.

### CODE REVIEW

File: src/tools/serialize_onnx/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ******************************************************************************
+
+add_executable(serialize_onnx serialize_onnx.cpp)
+add_dependencies(serialize_onnx ngraph)
+add_dependencies(serialize_onnx onnx_import_interface)
```
Comment (apwojcik) [inline src/tools/serialize_onnx/CMakeLists.txt]: This should not be required. If the build does not work without this target then the dependencies for ngraph are incorrectly set by ONNX.

File: src/tools/serialize_onnx/serialize_onnx.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        else if (arg == "-h" || arg == "--help")
+        {
+            help();
+            return 0;
+        }
+    }
+
+    ifstream f(input);
+    if (f)
```
Comment (apwojcik) [inline src/tools/serialize_onnx/serialize_onnx.cpp]: Instead, use `f.is_open()` to check if the file stream has an associated file.

File: src/tools/CMakeLists.txt [hunk 3]
```diff
@@ -16,3 +16,4 @@
 
 add_subdirectory(nbench)
 add_subdirectory(reserialize)
+add_subdirectory(serialize_onnx)
```
Comment (apwojcik) [inline src/tools/CMakeLists.txt]: Add the subdirectory ONLY if ONNX is enabled.

File: src/tools/serialize_onnx/CMakeLists.txt (hunk 1 above)
Comment (apwojcik) [inline src/tools/serialize_onnx/CMakeLists.txt]: Remove dependency on `onnx_import_interface`. By adding this you force build not having ONNX enabled to build the incomplete ONNX support in nGraph.

File: src/tools/serialize_onnx/serialize_onnx.cpp [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        else if (arg == "-h" || arg == "--help")
+        {
+            help();
+            return 0;
+        }
+    }
+
+    ifstream f(input);
+    if (f.is_open())
```
Comment (rkimballn1) [inline src/tools/serialize_onnx/serialize_onnx.cpp]: Better to just use `if (f)` which also tests for errors, not just open.

**Comment by arogowie-intel** [comment]
TEST NOW

**Comment by arogowie-intel** [comment]
TEST NOW

**Pull request merged**